src/nv_ingest/util/nim/helpers.py

```diff
@@ -107,29 +110,43 @@ def call_image_inference_model(client, model_name: str, image_data):
     return response
 
 
-def _call_image_inference_grpc_client(client, model_name: str, image_data):
+def _call_image_inference_grpc_client(client, model_name: str, image_data: np.ndarray):
     if image_data.ndim == 3:
         image_data = np.expand_dims(image_data, axis=0)
-    inputs = [grpcclient.InferInput("input", image_data.shape, "FP32")]
-    inputs[0].set_data_from_numpy(image_data.astype(np.float32))
+
+    if model_name in {"deplot", "paddle", "cached", "yolox"}:
+        inputs = [grpcclient.InferInput("input", image_data.shape, "FP32")]
+        inputs[0].set_data_from_numpy(image_data.astype(np.float32))
+    elif model_name == "doughnut":
+        inputs = [grpcclient.InferInput("input", image_data.shape, "UINT8")]
+        inputs[0].set_data_from_numpy(image_data.astype(np.uint8))
 
     outputs = [grpcclient.InferRequestedOutput("output")]
 
     try:
         result = client.infer(model_name=model_name, inputs=inputs, outputs=outputs)
-        return " ".join([output[0].decode("utf-8") for output in result.as_numpy("output")])
     except Exception as e:
         err_msg = f"Inference failed for model {model_name}: {str(e)}"
         logger.error(err_msg)
         raise RuntimeError(err_msg)
 
+    if model_name in {"deplot", "paddle", "cached", "yolox"}:
+        result = " ".join([output[0].decode("utf-8") for output in result.as_numpy("output")])
+    elif model_name == "doughnut":
+        result = [output.decode("utf-8") for output in result.as_numpy("output")]
+
+    return result
 
-def _call_image_inference_http_client(client, model_name: str, image_data):
-    base64_img = numpy_to_base64(image_data)
 
+def _call_image_inference_http_client(client, model_name: str, image_data: np.ndarray):
```

Comment:
I think we're getting a bit too specific in terms of data handling here. We probably want to standardize on a data format for each model and process it into that before we get to this function.

Small nitpick: please use base64_img(s) or base64_image(s), rather than base64_img and base64_images

Reply:
I cleaned up a bit in [c5b5d3e](https://github.com/NVIDIA/nv-ingest/pull/230/commits/c5b5d3e0f8b80a78d9fb68efca7f1cfeac834612) by basically grouping doughnut together with deplot. They have the same output format (for now). Note: The input/output format is not finalized and will likely change and actually be closer to the yolox output format so there will be more PRs to follow.